### ISSUE TITLE
Data leakage in Kaggle House Prices example

### ISSUE THREAD

**Issue by austinmw** [d2l-ai/d2l-en#843]
In _4.10.4. Data Preprocessing_, normalization is applied to `all_features`, therefore information from the test set is being leaked to the training set.

I believe you should instead fit the normalization to the training set, and then apply the fitted transform to both sets, for example:

```python
from sklearn.preprocessing import StandardScaler
scalar = StandardScaler()
X_train = scalar.fit_transform(X_train)
test = scalar.transform(test)
```

Ref: https://stackoverflow.com/questions/49444262/normalize-data-before-or-after-split-of-training-and-testing-data

I assume this may be intentional to simplify the example, but I think if that's the case a note should be made that this incurs leakage.

**Comment by austinmw** [comment]
Closing. After thinking about it, this is fine for a Kaggle competition.
